lib/bugsnag/payload.ex

```diff
@@ -116,13 +118,34 @@ defmodule Bugsnag.Payload do
         %{
           file: file,
           lineNumber: line_number,
-          inProject: Regex.match?(~r/^(lib|web)/, file),
+          inProject: in_project_fn.({module, function, args, file}),
           method: Exception.format_mfa(module, function, args),
           code: get_file_contents(file, line_number)
         }
     end)
   end
 
+  @spec get_in_project_fn(Keyword.t) :: (stack_frame -> boolean) when
+        stack_frame: {module, function :: atom, args :: list, String.t}
+  defp get_in_project_fn(options) do
+    always_false_fn = fn(_stack_frame) -> false end
+    case fetch_option(options, :in_project, nil) do
+      {mod, fun, state} ->
+        fn(stack_frame) -> apply(mod, fun, [stack_frame, state]) end
+      %Regex{} = re ->
+        fn({_m, _f, _a, file}) -> Regex.match?(re, file) end
+      re_str when is_binary(re_str) ->
+        case Regex.compile(re_str) do
+          {:ok, re} ->
+            fn({_m, _f, _a, file}) -> Regex.match?(re, file) end
+          {:error, _reason} ->
+            always_false_fn
```

Comment:
I think we should also issue a warning in this case — the user will want to know their regex was invalid.